joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -55,17 +55,13 @@ controller_interface::CallbackReturn JointTrajectoryController::on_init()
   try
   {
     // with the lifecycle node being initialized, we can declare parameters
-    auto_declare<std::vector<std::string>>("joints", joint_names_);
-    auto_declare<std::vector<std::string>>("command_interfaces", command_interface_types_);
-    auto_declare<std::vector<std::string>>("state_interfaces", state_interface_types_);
-    auto_declare<double>("state_publish_rate", 50.0);
-    auto_declare<double>("action_monitor_rate", 20.0);
-    auto_declare<bool>("allow_partial_joints_goal", allow_partial_joints_goal_);
-    auto_declare<bool>("open_loop_control", open_loop_control_);
-    auto_declare<bool>(
+    joint_names_ = auto_declare<std::vector<std::string>>("joints", joint_names_);
+    command_interface_types_ = auto_declare<std::vector<std::string>>("command_interfaces", command_interface_types_);
+    state_interface_types_ = auto_declare<std::vector<std::string>>("state_interfaces", state_interface_types_);
+    allow_partial_joints_goal_ = auto_declare<bool>("allow_partial_joints_goal", allow_partial_joints_goal_);
+    open_loop_control_ = auto_declare<bool>("open_loop_control", open_loop_control_);
+    allow_integration_in_goal_trajectories_ = auto_declare<bool>(
       "allow_integration_in_goal_trajectories", allow_integration_in_goal_trajectories_);
-    auto_declare<double>("constraints.stopped_velocity_tolerance", 0.01);
-    auto_declare<double>("constraints.goal_time", 0.0);
```

Comment:
As far as I can tell, constraints.goal_time isn't used anywhere. Same for `stopped_velocity_tolerance`, `state_publish_rate`, and `action_monitor_rate`.

Reply:
yeah, these params I think are pretty much just used in the tolerances.hpp file https://github.com/ros-controls/ros2_controllers/blob/master/joint_trajectory_controller/include/joint_trajectory_controller/tolerances.hpp#L103